src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -941,13 +945,13 @@ public override BoundNode VisitAnonymousObjectCreationExpression(BoundAnonymousO
                             receiverSlot = GetOrCreateSlot(implicitReceiver);
                         }
 
-                        TrackNullableStateForAssignment(arguments[i], property, GetOrCreateSlot(property, receiverSlot), arguments[i], this.State.ResultIsNotNull);
+                        TrackNullableStateForAssignment(arguments[i], property, GetOrCreateSlot(property, receiverSlot), arguments[i], this.State.ResultType);
                     }
                 }
 
                 if (_trackExceptions) NotePossibleException(node);
 
-                this.State.ResultIsNotNull = true;
+                this.State.ResultType = TypeSymbolWithAnnotations.Create(node.Type);
```

Comment:
>node.Type [](start = 73, length = 9)

Is it possible that nullability of nested type references is not accurate?

Reply:
Yes.